wurst/lib/Lodash.wurst

```diff
@@ -305,6 +315,199 @@ public function each<T>(VoidFunction<T> func, LinkedList<T> list)
 public function LinkedList<T>.each<T>(VoidFunction<T> func)
     each(func, this)
 
+public function zip<A, B>(LinkedList<A> a, LinkedList<B> b) returns IterableMap<A, B>
+    let map = new IterableMap<A, B>()
+    let iterA = a.iterator()
+    let iterB = b.iterator()
+    while iterA.hasNext()
+        let bVal = iterB.hasNext()
+            ? iterB.next()
+            : null
+        map.put(iterA.next(), bVal)
```

Comment:
Is this how `zip` works in Lodash? I haven't used it before but the Python version ends with the shorter iterable, rather than automatically padding it with nulls. Also seems weird to have the operation not be commutative.

Reply:
Indeed that is how `zipObject` works in Lodash. I can change it though.

Reply:
@jonathanmclaus yeah I guess I've implemented `zipObject`. I'll rename and add `zip`.

Reply:
Yeah, I'm an idiot and was going off the name. I think it's fine to map to nulls in that case.

Reply:
@jonathanmclaus updated